Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift

```diff
@@ -75,200 +91,1089 @@ public class ComponentViewController: UIView {
             if componentType[i] == "image" {
-                if pickedImg.count == 0 {
-                    cellHeight.append(160)
+                if imageSelectionCount[i].count == 0 {
+                    if selectedPicture[i].count == 0 {
+                        cellHeight.append(150)
+                    } else {
+                        cellHeight.append(270)
+                    }
                 } else {
                     cellHeight.append(270)
                 }
             }
             if componentType[i] == "text" {
-                cellHeight.append(90)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "multiSelect" {
-                cellHeight.append(190)
+                cellHeight.append(componentTableViewCellHeight[i] + 20)
             }
             if componentType[i] == "dropdown" {
-                cellHeight.append(100)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "textarea" {
-                cellHeight.append(190)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "date" {
-                cellHeight.append(100)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "signature" {
-                cellHeight.append(260)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "block" {
                 cellHeight.append(50)
             }
             if componentType[i] == "number" {
-                cellHeight.append(100)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "chart" {
-                cellHeight.append(380)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
             }
             if componentType[i] == "table" {
-                cellHeight.append(tableHeight)
+                cellHeight.append(componentTableViewCellHeight[i] + 10)
+            }
+            if componentType[i] == "richText" {
+                cellHeight.append(componentTableViewCellHeight[i])
             }
         }
         return cellHeight[indexPath.row]
     }
+    
+    //MARK: - Image Field
+    func configureImageFieldCell(tableView: UITableView, i: Int, value: ValueUnion?) {
+        let image = Image()
+        image.index = i
+        image.saveDelegate = self
+        image.uploadButton.tag = i
+        image.selectedImage = selectedPicture
+        image.pickedSingleImg = pickedSinglePicture
+        
+        imageSelectionCount[i].removeAll()
+        image.selectedImage[i].removeAll()
+        image.pickedSingleImg[i].removeAll()
+        
+        // Image value based on indexPath from JoyDoc
+        switch value {
+        case .string(_): break
```

Comment:
I think yo just need to add indentation here. Also can you explain this? Why so many `case - breaks` that do nothing other than just use `break`

Reply:
Values of the components are coming in different cases. So, to handle them these switch cases are created and 'case' label in a 'switch' must have at least one executable statement that is break is used in that case where value is not coming for that component.